### ISSUE TITLE
Add ST_Accessibility

### ISSUE THREAD

**Issue by agouge** [orbisgis/gdms-topology#26]
Add an SQL function with the following behavior:
- Inputs:
  - `edges`
  - >= 2 destination nodes
- Output: The `nodes` table with the id of the closest destination node (and optionally its distance to this node)
##### Calculation

A naive implementation would use Dijkstra to calculate the distance from each source node to all of the destination nodes and take the min. In practice, this will be too slow.

A better method is the following: 
1. Pick a source node and execute Dijkstra until one of the destination nodes is encountered. This is the closest one. Store its id and distance.
2. For each node on the shortest path from the source node to the newly found destination node, store the id of and distance to the destination.
3. Repeat steps 1-2, except that at each step along Dijkstra, check if the node being examined already has a destination node id. If so, then store the same id, and store the distance to the destination of every node along the shortest path back to the source.

_Note_: This would be rather easy to implement in undirected networks starting from each of the destination nodes (using the calculateShortestPathsFromNode method and an adapted NodeBetweennessInfo). In directed networks, however, we have to start from each of the source nodes.

**Comment by agouge** [comment]
An even better method (for directed networks) is the following:
1. Construct a directed network, then reverse all edge orientations.
2. Calculate the shortest paths **from each destination** to every other source.
3. Reverse the orientation again.

The advantage to this is that we have to run Dijkstra only |destinations| (once from each destination) times, rather than |V| times (once from every node).
@ebocher @gpetit

**Comment by agouge** [comment]
This table function will be called

``` mysql
ST_Accessibility(edges, destinations)
```

and will return each node with its closest destination and the distance to that destination.

**Comment by agouge** [comment]
We will also need to check to make sure that all sources/destinations are contained in the graph. If not, then raise an Exception with the missing node's id.

**Comment by ebocher** [comment]
I'm not in favour to add a specific function st_accessibility. Why not a st_shortestpathlength with 

st_shortestpathlength(edges, destinations, costColumn, graphType) ?

We must complete this discussion asap.

@gpetit

**Comment by agouge** [comment]
@ebocher `ST_Accessibility` serves a fundamentally different purpose than `ST_ShortestPathLength`. See #58.
